Editor/Deployment/DeploymentStatusPoller.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Threading.Tasks;
+using AmazonGameLiftPlugin.Core.DeploymentManagement.Models;
+using AmazonGameLiftPlugin.Core.Shared;
+
+namespace AmazonGameLift.Editor
+{
+    internal class DeploymentStatusPoller
+    {
+        public event Action InfoUpdated;
+
+        private readonly CoreApi _coreApi;
+        private bool _isWaiting;
+
+        public DeploymentStatusPoller(CoreApi coreApi)
+        {
+            _coreApi = coreApi;
+        }
+
+        public async Task<Response> WaitUntilDone(DeploymentSettings settings)
```

Comment:
Why can't you use the `WaitForCurrentDeployment` method in DeploymentSettings? Why is there a need for a separate Poller?

Reply:
I would prefer we reuse existing functionality unless strongly required otherwise.

Reply:
Swapped to using existing functionality